transport/transport.go

```diff
@@ -330,12 +330,20 @@ func NewServerTransport(protocol string, conn net.Conn, maxStreams uint32, authI
 	return newHTTP2Server(conn, maxStreams, authInfo)
 }
 
+// DefaultDialerFn is the default dialer.
+func DefaultDialerFn(addr string, timeout time.Duration, cancel <-chan struct{}) (net.Conn, error) {
```

Comment:
I've moved these functions out of `transport` and into `grpc`. Is that a reasonable compromise?